libs/search-widget/src/core/stores/answers.store.ts

```diff
@@ -25,22 +32,30 @@ export const currentAnswer = answerState.writer<Partial<Answer>>(
 );
 
 export const firstAnswer = answerState.reader((state) =>
-  state.isStreaming ? state.currentAnswer : state.dialog[0]?.answer || EMPTY_ANSWER,
-);
-export const lastOrCurrentAnswer = answerState.reader((state) =>
-  state.isStreaming ? state.currentAnswer : state.dialog[state.dialog.length - 1]?.answer || EMPTY_ANSWER,
+  state.dialog.length === 0 && state.isStreaming
+    ? { ...state.currentAnswer, incomplete: true }
+    : state.dialog[0]?.answer || EMPTY_ANSWER,
 );
 
 export const dialog = answerState.writer<DialogEntry[], { question: string; answer: Answer; reset: boolean }>(
-  (state) => state.dialog,
+  (state) =>
+    state.isStreaming
+      ? [...state.dialog, { question: state.currentQuestion, answer: { ...state.currentAnswer, incomplete: true } }]
+      : state.dialog,
   (state, params) => ({
     ...state,
     dialog: params.reset
       ? [{ question: params.question, answer: params.answer }]
       : [...state.dialog, { question: params.question, answer: params.answer }],
+    currentQuestion: '',
     currentAnswer: EMPTY_ANSWER,
     isStreaming: false,
   }),
 );
 
+export const resetDialog = answerState.writer<void>(
+  () => undefined,
+  (state) => ({ ...state, dialog: state.dialog.slice(0, 1) }),
+);
```

Comment:
why not creating a simple function for this? a writer seems a bit overkill here

Reply:
I don't get it. How could we change the state if it is not through a `writer`?

Reply:
you can use the `update` function of your state. See in `annotations` store: we have our store, we have a few writers for the things we read and write, and we have a few functions allowing to just update specific things of the store.